### PR TITLE
Make an Abstraction package

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
As a first step for API clean up I separated code into two packages - `OpenCensus` and `OpenCensus.Abstractions`. `Abstraction` package will contain lightweight interfaces only.

### CODE REVIEW

File: src/OpenCensus.Abstractions/Utils/Collections.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                return "null";
+            }
+
+            StringBuilder sb = new StringBuilder();
+            foreach (var kvp in dict)
+            {
+                sb.Append(kvp.Key.ToString());
+                sb.Append("=");
+                sb.Append(kvp.Value.ToString());
```
Comment (lmolkova) [inline src/OpenCensus.Abstractions/Utils/Collections.cs]: value could be null

File: src/OpenCensus.Abstractions/Utils/Collections.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                sb.Append(" ");
+            }
+
+            return sb.ToString();
+        }
+
+        public static string ToString<T>(IList<T> list)
```
Comment (lmolkova) [inline src/OpenCensus.Abstractions/Utils/Collections.cs]: Do we need this function at all?

list == null ? null : string.Join(" ", list)?

File: src/OpenCensus.Abstractions/Utils/Collections.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+namespace OpenCensus.Utils.Abstractions
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+
+    internal static class Collections
+    {
+        public static string ToString<TKey, TValue>(IDictionary<TKey, TValue> dict)
+        {
```
Comment (lmolkova) [inline src/OpenCensus.Abstractions/Utils/Collections.cs]: do we need this method:

string.Join(" ", dict.Select(kvp => $"{kvp.Key}={kvp.Value}")). 
`$"{kvp.Key}={kvp.Value}"` could be replaced by string.Concat for a bit better performance

**Comment by SergeyKanzhelev** [comment]
Thank you! That file is just a copy of existing one. Comments make sense, but I'd address them later

**Pull request merged**